staging/src/k8s.io/apiserver/pkg/util/feature/feature_gate.go

```diff
@@ -157,6 +159,28 @@ func (f *featureGate) Set(value string) error {
 	return nil
 }
 
+// SetFromMap stores flag gates for known features from a map[string]bool or returns an error
+func (f *featureGate) SetFromMap(m map[string]bool) error {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+
+	for k, v := range m {
```

Comment:
Does order matter in our implementation?

Reply:
Since you can only set a given key once from a single map, I'm not sure it matters for `SetFromMap`.

With `Set`, which takes a string, I could see a bunch of parameters being concatenated while someone builds a command line, and some later ones overriding some previous ones. But that matters for CLI only.

I don't think order would matter for anything else. Feature gates are stored internally as a map, so any ordering information would have to be considered by `Set`, which it doesn't currently do, even with the special feature handlers (previously set fields are ignored by `setUnsetAlphaGates`, and subsequently set fields override, so `allAlphaGate` has the same effect no matter where it appears in the iteration order).

Reply:
I'm fairly certain if you removed the defaulting, you would get a nil pointer dereference and that's why that is there.